ts/Core/Series/DataLabel.ts

```diff
@@ -492,14 +507,35 @@ namespace DataLabel {
      * Create the SVGElement group for dataLabels
      * @private
      */
-    function initDataLabelsGroup(this: Series): SVGElement {
-        return this.plotGroup(
+    function initDataLabelsGroup(
+        this: Series,
+        index: number,
+        dataLabelsOptions?: DataLabelOptions
+    ): SVGElement {
+        fireEvent(
+            this,
+            'initDataLabelsGroup',
+            { index, zIndex: dataLabelsOptions?.zIndex || 6 }
```

Comment:
Sooner or later someone will try to set `zIndex: 0` and wonder why it is not working
```suggestion
            { index, zIndex: dataLabelsOptions?.zIndex ?? 6 }
```